unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Utils/ECSComponentsUtils.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using DCL;
+using DCL.Configuration;
+using DCL.Helpers;
+using DCL.Models;
+using UnityEngine;
+
+public static class ECSComponentsUtils
+{
+    public static MeshesInfo GenerateMeshInfo(IDCLEntity entity, Mesh mesh, GameObject gameObject,bool visible, bool withCollisions, bool isPointerBlocker)
+    {
+        MeshesInfo meshesInfo = new MeshesInfo();
+        meshesInfo.innerGameObject = gameObject;
+        meshesInfo.meshRootGameObject = gameObject;
+        MeshFilter meshFilter = gameObject.AddComponent<MeshFilter>();
+        MeshRenderer meshRenderer = gameObject.AddComponent<MeshRenderer>();
+        meshRenderer.sharedMaterial = Utils.EnsureResourcesMaterial("Materials/Default");
+        Renderer[] renderers = new Renderer[] { meshRenderer };
+        
+        meshFilter.sharedMesh = mesh;
+
+        // We should remove this relation in the future, the entity shouldn't know about the mesh
+        entity.meshesInfo = meshesInfo;
+        
+        UpdateRenderer(entity,gameObject, renderers, visible, withCollisions, isPointerBlocker);
+        meshesInfo.UpdateRenderersCollection();
+        return meshesInfo;
+    }
+    
+    public static void UpdateRenderer(IDCLEntity entity,GameObject meshGameObject, Renderer[] renderers,bool visible, bool withCollisions, bool isPointerBlocker)
+    {
+        ConfigureVisibility(meshGameObject, visible,renderers);
+        
+        CollidersManager.i.ConfigureColliders(meshGameObject, withCollisions, false, entity, CalculateCollidersLayer(withCollisions,isPointerBlocker));
+    }
+    
+    public static void ConfigureVisibility(GameObject meshGameObject, bool shouldBeVisible, Renderer[] meshRenderers = null)
+    {
+        if (meshGameObject == null)
```

Comment:
This is doing way too many things and it can become a bottleneck (all these `GetComponentsInChildren` are gonna hurt performance).

Shouldn't most of this be part of the ECSComponents (probably the handlers) responsibility?
`MaterialTransitionController` => `GLTFShape`
`PointerEvent` => Whateversets the collision (shapes ??)

These ECSComponents should know already most of that stuff and avoid the GetComponents calls

Reply:
Yep, the problem here is with the GLTFShape, but we can tackle that in the future.

For now, I just removed the MaterialTransitioController, since it is only for the GLTF and not primitives and also have refactored the code a bit to avoid the GetComponents as much as possible, the only thing that is left here is the MeshCollider, I have left a TODO for that so when we have all the primitive components we can tackle this problem with a better approach